portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -32,38 +39,74 @@
+    PreparedQuery results = datastore.prepare(query);
+
+    // Get the max amount of comments with id, text, and timestamp.
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String text = (String) entity.getProperty("text");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Comment comment = new Comment(id, text, timestamp);
+      comments.add(comment);
+      if(comments.size() >= maxCommentsObtained){
+        break;
+      }
+    }
+
+    Gson gson = new Gson();
+
     response.setContentType("application/json;");
-    response.getWriter().println(json);
+    response.getWriter().println(gson.toJson(comments));
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get the input from the form.
-    String text = getParameter(request, "text-input", "");
-
-    // Respond with the result.
-    comments.add(text);
-    response.setContentType("text/html;");
-    for (int commentIndex = 0; commentIndex < comments.size(); commentIndex++){
-      response.getWriter().println(comments.get(commentIndex));
-    }
+    // Get the comment text input from the form.
+    String text = request.getParameter("text-input");
+    long timestamp = System.currentTimeMillis();
+
+    // Creates data in Datastore with the text as a comment property.
+    Entity taskEntity = new Entity("Comment");
+    taskEntity.setProperty("text", text);
+    taskEntity.setProperty("timestamp", timestamp);
+
+    // Put newly created Entity.
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(taskEntity);
 
     // Redirect back to the HTML page.
     response.sendRedirect("/index.html");
   }
 
-  /**
-   * Returns the request parameter.
-   * @return the request parameter, or the default value if the parameter
-   *         was not specified by the client.
-   */
-  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
-    String value = request.getParameter(name);
-    if (value == null) {
-      return defaultValue;
+  /** Returns the max number of comments to display, or -1 if the choice was invalid. */
+  private int getMaxComments(HttpServletRequest request) {
+    // Get the number of comments to display from the form.
+    String stringMaxComments = request.getParameter("max-comments");
+
+    // Convert the input to an int.
+    int maxComments = 1;
+    try {
+      maxComments = Integer.parseInt(stringMaxComments);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + maxComments);
+      return -1;
     }
-    return value;
+
+    // Check that the input is between 1 and 3.
+    if (maxComments < 1 || maxComments > 3) {
+      System.err.println("The maximum number of comments chosen is out of range: " + maxComments);
```

Comment:
You might also want to let the user know why they're getting the error so they don't keep trying random numbers and get frustrated :D

```suggestion
      System.err.println("Error: Maximum number of comments must be 1-3. Value entered: " + maxComments);
```